articles/feature-engineering-in-machine-learning/index.md

```diff
@@ -0,0 +1,167 @@
+
+- **Classification models** – classification tasks involve predicting the type of an object. The types of objects are finite. These models handle classification tasks. Examples of these models include; [Naïve Bayes](https://www.machinelearningplus.com/predictive-modeling/how-naive-bayes-algorithm-works-with-example-and-full-code/), [Decision Tree](http://mines.humanoriented.com/classes/2010/fall/csci568/portfolio_exports/lguo/decisionTree.html#:~:text=Decision%20Tree%20Classifier%20is%20a%20simple%20and%20widely,an%20answer%2C%20a%20follow-up%20question%20is%20asked%20), [Support Vector Machine](https://towardsdatascience.com/support-vector-machine-introduction-to-machine-learning-algorithms-934a444fca47) among others. A brief definition of these classification models can be found in my [previous article](/engineering-education/automated-fake-news-detection/)
+
+- **Regression models** – unlike classification models, these tasks give output variables that take continuous values. An example of such a model is a linear regression model.
+
+#### Machine Learning Categories
+There are three types of machine learning:
+
+- **Supervised learning** - an example of this is a student being supervised by a teacher. The student is trying to solve a problem in class. The teacher oversees that the student solves the problem correctly. For a model, this happens through use of labeled data. The labeled data, as input, guides the model. Classification and regression models are supervised learning algorithms.
+
+- **Reinforcement learning** - learning is through a trial-and-error approach. An agent/machine performs actions and is either rewarded or learns from errors.
+
+- **Unsupervised learning** - simply the opposite of supervised learning. There is no clean, complete, or labelled dataset provided. While supervised learning output relies on labeled data, unsupervised output is based on the patterns identified in input data. Input data has no pre-existing labels.
+
+We are now acquainted with machine learning. But, in order to understand what feature engineering is, we need to have an overview of the entire machine learning process.
+
+#### Machine Learning Process
+It is important to understand the machine learning lifecycle. These are more or less the steps of the machine learning process:
+
+- **Collecting data**- data is gathered from various sources. In any machine learning workflow, this is the first step. It is a very important step since the right data translates to better results.
+
+- **Preparing that data**- to get some form of meaning out of data, a number of processes take place in this stage. Datasets are evaluated for missing values, outliers, trends, and so on. Oddities in data are rectified. Data is cleaned. It is worth noting that the feature engineering process takes place here. I argue this to be the most important step in the lifecycle.
+
+- **Choosing a model**- this step involves selection of a model best suited to the prepared data.
```

Comment:
As a reader, I want to know how do I choose the best model? So, consider adding information about how multiple machine learning models are compared. Adding information about model comparisons and cross-validation would deliver more value.